fetch-server/search.py

```diff
@@ -0,0 +1,61 @@
+from models.route import Route, RouteAdmin
+from flask_restful import reqparse, abort, Api, Resource, fields, marshal_with
+from geoalchemy2.shape import to_shape
+import utils
+
+api = utils.get_api()
+
+# GET parameters for the search
+parser = reqparse.RequestParser(trim=True, bundle_errors=True)
+parser.add_argument('id'             , type=int,  store_missing=False)
+parser.add_argument('name'           , type=str,  store_missing=False)
+parser.add_argument('address'        , type=str,  store_missing=False)
+parser.add_argument('is_shade'       , type=bool, store_missing=False)
+parser.add_argument('is_water'       , type=bool, store_missing=False)
+parser.add_argument('is_garbage_can' , type=bool, store_missing=False)
+parser.add_argument('is_poop_bag'    , type=bool, store_missing=False)
+
+# JSON response
+# { 'results' : [ route1, route2, ...] }
+# where each route has the following fields:
+
+class Coordinates(fields.Raw):
+    def format(self, wkb):
+        coords = to_shape(wkb).coords[:]
+        return map(lambda (x,y): [x,y], coords)
+
+route_fields = {
+    'id'             : fields.Integer,
+    'name'           : fields.String,
+    'address'        : fields.String,
+    'is_shade'       : fields.Boolean,
+    'is_water'       : fields.Boolean,
+    'is_garbage_can' : fields.Boolean,
+    'is_poop_bag'    : fields.Boolean,
+    'coodinates'     : Coordinates(attribute='path')
+}
+
+# filter & respond to the search query
+class RouteSearch(Resource):
+    @marshal_with(route_fields, envelope='results')
+    def get(self):
```

Comment:
This search is actually off. For some reason when I search `?is_poop_bag=False` the backend is converting it to True. It is doing the same with the wildlife endpoints that I am creating. Ignore it for now though until we know the fix

Reply:
In order to make it false, I had to give it `?is_poop_bag` by itself.